dvc/external_repo.py

```diff
@@ -123,6 +124,22 @@ def _set_cache_dir(self):
 
         self.cache.local.cache_dir = cache_dir
 
+    def _set_relative_upstream(self):
+        # check if the URL is local and if a default relative-path
+        # remote is present, if so, make the URL relative to the
+        # original repo
+        if os.path.isdir(self.url):
+            remote_name = self.config["core"].get("remote")
+            if remote_name:
+                src_repo = Repo(self.url)
+                remote_cfg = self.config["remote"][remote_name]
+                old_remote_url = src_repo.config["remote"][remote_name]["url"]
+                if remote_cfg["url"] != old_remote_url:
+                    remote_cfg["url"] = os.path.join(
+                        src_repo.root_dir, old_remote_url
+                    )
+                src_repo.close()
```

Comment:
You might be leaking repo resources if any error happens above. Better to do the same thing as in `_set_upstream` down below and put `repo.close()` into `finally` block.